### PR TITLE
Feature/gtm major compatibility

### PR DESCRIPTION
**Pull request by icazevedo**
#### What problem is this solving?

This PR makes the search result compatible with the new major of the GTM app.

#### How to test it?

[Workspace](https://icarogtm--storecomponents.myvtex.com/)

#### Related to / Depends on

Depends on [https://github.com/vtex-apps/product-summary/pull/308](https://github.com/vtex-apps/product-summary/pull/308)
Depends on [https://github.com/vtex-apps/product-summary-context/pull/18](https://github.com/vtex-apps/product-summary-context/pull/18)

### CODE REVIEW

File: react/package.json [hunk 1]
```diff
@@ -27,42 +27,43 @@
...
-    "vtex.rich-text": "http://vtex.vtexassets.com/_v/public/typings/v1/vtex.rich-text@0.11.2/public/@types/vtex.rich-text",
+    "vtex.device-detector": "http://vtex.vtexassets.com/_v/public/typings/v1/vtex.device-detector@0.2.6/public/@types/vtex.device-detector",
+    "vtex.flex-layout": "http://vtex.vtexassets.com/_v/public/typings/v1/vtex.flex-layout@0.16.0/public/@types/vtex.flex-layout",
+    "vtex.format-currency": "http://vtex.vtexassets.com/_v/public/typings/v1/vtex.format-currency@0.3.0/public/@types/vtex.format-currency",
+    "vtex.native-types": "http://vtex.vtexassets.com/_v/public/typings/v1/vtex.native-types@0.7.5/public/@types/vtex.native-types",
+    "vtex.pixel-manager": "http://vtex.vtexassets.com/_v/public/typings/v1/vtex.pixel-manager@1.8.0/public/@types/vtex.pixel-manager",
+    "vtex.product-list-context": "http://vtex.vtexassets.com/_v/public/typings/v1/vtex.product-list-context@0.3.1/public/@types/vtex.product-list-context",
+    "vtex.product-summary": "https://icarogtm--storecomponents.myvtex.com/_v/private/typings/linked/v1/vtex.product-summary@2.71.2+build1620066564/public/@types/vtex.product-summary",
```
Comment (icazevedo) [inline react/package.json:52]: Change this before publishing

File: react/package.json [hunk 2]
```diff
@@ -27,42 +27,43 @@
...
+    "vtex.render-runtime": "http://vtex.vtexassets.com/_v/public/typings/v1/vtex.render-runtime@8.128.4/public/@types/vtex.render-runtime",
+    "vtex.responsive-values": "http://vtex.vtexassets.com/_v/public/typings/v1/vtex.responsive-values@0.4.2/public/@types/vtex.responsive-values",
+    "vtex.rich-text": "http://vtex.vtexassets.com/_v/public/typings/v1/vtex.rich-text@0.13.0/public/@types/vtex.rich-text",
     "vtex.search-page-context": "http://vtex.vtexassets.com/_v/public/typings/v1/vtex.search-page-context@0.2.0/public/@types/vtex.search-page-context",
+    "vtex.search-result": "https://icarogtm--storecomponents.myvtex.com/_v/private/typings/linked/v1/vtex.search-result@3.98.3+build1620066784/public/@types/vtex.search-result",
```
Comment (icazevedo) [inline react/package.json:57]: Change this before publishing

File: react/hooks/useSearchTitle.ts [hunk 3]
```diff
@@ -0,0 +1,110 @@
+import { useContext, useMemo } from 'react'
+
+// This is here so this JS file can be imported
```
Comment (klzns) [inline react/hooks/useSearchTitle.ts:3]: Thanks for removing ramda 🙏

**Review by klzns — APPROVED** [review]

**Review by igorbrasileiro — APPROVED** [review]

**Review by hiagolcm — APPROVED** [review]

File: react/package.json (hunk 1 above)
Comment (icazevedo) [reply react/package.json]: Also change the shelf import

File: react/components/GalleryItem.tsx [hunk 4]
```diff
@@ -17,6 +17,8 @@ interface GalleryItemProps {
   summary: any
   /** Item position in the gallery */
   position: number
+  /** Name of the list of products */
```
Comment (victorhmp) [inline react/components/GalleryItem.tsx:20]: Might be useful to expand this just a little to include the purpose of this name. Like "This name will be used when sending pixel events" for example

**Review by victorhmp — APPROVED** [review]

**Pull request merged**